app/presenters/completions_presenter.rb

```diff
@@ -42,21 +42,50 @@ def ial2_requested?
     @ial2_requested
   end
 
+  def sp_name
+    @sp_name ||= current_sp.friendly_name || sp.agency&.name
+  end
+
   def heading
     if ial2_requested?
-      I18n.t('titles.sign_up.completion_ial2', app_name: APP_NAME)
+      I18n.t('titles.sign_up.completion_ial2', sp: sp_name)
     elsif first_time_signing_in?
-      I18n.t('titles.sign_up.completion_first_sign_in', app_name: APP_NAME)
-    elsif completion_context == :consent_expired
+      I18n.t('titles.sign_up.completion_first_sign_in', sp: sp_name)
+    elsif consent_has_expired?
       I18n.t('titles.sign_up.completion_consent_expired')
     elsif completion_context == :new_attributes
-      sp_name = current_sp.friendly_name || sp.agency&.name
       I18n.t('titles.sign_up.completion_new_attributes', sp: sp_name)
     else
       I18n.t('titles.sign_up.completion_new_sp')
     end
   end
 
+  def intro
```

Comment:
Would be good to have specs for this. There are `#heading` specs we could draw from.